mlcomp/parallelm/pipeline/dag.py

```diff
@@ -112,10 +112,13 @@ def run_connected_pipeline(self, system_conf, engine_info):
             sys.stdout.flush()
 
             runtime_in_sec = time.time() - start
-            if data_objs and type(data_objs) is not list:
-                raise MLCompException("Invalid returned data type from component! It should be a list! "
+            if data_objs is None:
```

Comment:
But `data_objs` can be None or empty list! Assuming we support components that are not connected in a pipeline or sink components.

Reply:
I agree with Zohar. If the component is supposed to return something then we should fail. But if it is a Sink, without output we should not